packages/playwright-test/src/runner/uiMode.ts

```diff
@@ -117,16 +112,23 @@ class UIMode {
         await queue;
       },
 
-      onclose: () => exitPromise.resolve(),
+      onclose: () => { },
     };
-    await showTraceViewer([], 'chromium', {
+    const openOptions: OpenTraceViewerOptions = {
       app: 'uiMode.html',
       headless: isUnderTest() && process.env.PWTEST_HEADED_FOR_TEST !== '1',
       transport: this._transport,
       host: options.host,
       port: options.port,
-      openInBrowser: options.host !== undefined || options.port !== undefined,
-    });
+    };
+    const exitPromise = new ManualPromise<void>();
+    if (options.host !== undefined || options.port !== undefined) {
+      await openTraceInBrowser([], openOptions);
+      process.on('SIGINT', () => exitPromise.resolve());
```

Comment:
This doesn't really work nicely with running tests. Perhaps not do this and file a bug instead? I think missing teardown is worse than interfering with test run.

Reply:
https://github.com/microsoft/playwright/issues/23907 for further reference.